packages/start/routes.js

```diff
@@ -53,27 +53,26 @@ export async function getRoutes({
       if (!exports.includes("default")) {
         return;
       }
-    }
-
-    if (!parentRoute) {
-      routesList.push({
-        src: src,
-        _id: id,
-        path: toPath(id) || "/",
-        componentSrc: src,
-        type: "PAGE",
-        dataSrc: data[full] ? data[full] : exports.includes("data") ? src + "?data" : undefined
-        // methods: ["GET", ...(exports.includes("action") ? ["POST", "PATCH", "DELETE"] : [])]
-        // actionSrc: exports.includes("action") ? src + "?action" : undefined,
-        // loaderSrc: exports.includes("loader") ? src + "?loader" : undefined
-      });
-    } else {
-      processRoute(
-        parentRoute.children || (parentRoute.children = []),
-        src,
-        id.slice(parentRoute._id.length),
-        full
-      );
+      if (!parentRoute) {
+        routesList.push({
+          src: src,
+          _id: id,
+          path: toPath(id) || "/",
+          componentSrc: src,
+          type: "PAGE",
+          dataSrc: data[full] ? data[full] : exports.includes("data") ? src + "?data" : undefined
+          // methods: ["GET", ...(exports.includes("action") ? ["POST", "PATCH", "DELETE"] : [])]
+          // actionSrc: exports.includes("action") ? src + "?action" : undefined,
+          // loaderSrc: exports.includes("loader") ? src + "?loader" : undefined
+        });
+      } else {
+        processRoute(
+          parentRoute.children || (parentRoute.children = []),
+          src,
+          id.slice(parentRoute._id.length),
+          full
+        );
+      }
```

Comment:
putting this inside here.. but prevent non-`.tsx,.ts,.jsx,.js` routes to be includes... eg. `.mdx`, the fix is that dataSrc shouldn't use the exports.. and instead be what I pointed out earlier